iotdb-core/datanode/src/main/java/org/apache/iotdb/db/queryengine/execution/operator/process/TransformOperator.java

```diff
@@ -141,16 +149,31 @@ protected void initTransformers(
               .buildLayerMemoryAssigner()
               .bindInputLayerColumnIndexWithExpression()
               .buildResultColumnPointReaders()
-              .getOutputPointReaders();
+              .getOutputReaders();
     } finally {
       UDFManagementService.getInstance().releaseLock();
     }
   }
 
   protected YieldableState iterateAllColumnsToNextValid() throws Exception {
+    if (transformers.length == 1) {
+      if (outputColumns[0] == null) {
+        YieldableState state = transformers[0].yield();
+        if (state != YieldableState.YIELDABLE) {
+          return state;
+        }
+
+        Column[] columns = transformers[0].current();
+        TsBlock block = new TsBlock((TimeColumn) columns[1], columns[0]);
```

Comment:
Why not use API wrapBlocksWithoutCopy🤔？

Reply:
With this API:
```java
int count = columns[0].getPositionCount();
Column[] valueColumns = { columns[0] };
TsBlock block = TsBlock.wrapBlocksWithoutCopy(count, (TimeColumn) columns[1], valueColumns);
```
Without this API:
```java
TsBlock block = new TsBlock((TimeColumn) columns[1], columns[0]);
```
The code hasn't gotten any more elegant.